### DISCUSSION TITLE
Trying to pre-build cache in Mikro-Orm V6

### DISCUSSION THREAD

**Discussion by Sundhar-Karunanidi-Bjss** [mikro-orm/mikro-orm#5152] [General]
I am currently using the MikroORMModule.forrootasync() (@microorm/nestjs) to configure Mikro-ORM dynamically. I am trying to generate a pre built cache but I keep getting an error where the code is looking for config. Is there a way to build the cache using this MikroORMModule?

**Comment by B4nan** [comment]
You need to have a CLI config, the ORM does not understand nest-specific things like the `forRoot` call. There you can import your config and use it in the `forRoot` to ahve a single source of truth.

**Reply by Sundhar-Karunanidi-Bjss** [reply]
So would this work if I am currently dynamically creating the config? or am I thinking about that in the wrong way 

```
MikroOrmModule.forRootAsync({
      imports: [ConfigModule],
      inject: [ConfigService],
      useFactory: (configService: ConfigService) => ({
        entities: ['./dist/src/domain'],
        entitiesTs: ['./src/domain'],
        user: configService.get<string>(ConfigNames.POSTGRES_USER, ''),
        password: configService.get<string>(ConfigNames.POSTGRES_PASSWORD, ''),
        dbName: configService.get<string>(ConfigNames.POSTGRES_DB, ''),
        schema: 'products',
        driver: PostgreSqlDriver,
        metadataProvider: TsMorphMetadataProvider,
        metadataCache: {
          enabled: true,
          adapter: GeneratedCacheAdapter,
          options: { data: require('./temp/metadata.json') },
        },
        debug: true,
      }),
    })
``` 
right now my app.module.ts looks like this.

**Reply by B4nan** [reply]
As I said, you need cli config. That's the file called `mikro-orm.config.ts` (depends on your setup, usually should be in src folder). 

https://mikro-orm.io/docs/quick-start#setting-up-the-commandline-tool

**Reply by gusgorman2** [reply]
As far as I understand this as we are dynamically generating some of our config via services in the Nestjs container (ie to retrieve user, password, and dbName from environment variables) we can't use mikro-orm.config.ts for our application.

So does this mean we can't pre-build the cache? 

Or would we have to find some solution where we duplicate the config, ie have a config file for generating the cache and code config for the app? I'm also assuming pre-building the cache only needs a subset of config?

Sorry for all the questions !

**Reply by B4nan** [reply]
You need to have a CLI config to use the CLI. Alternatively, create a script instead.

**Reply by Sundhar-Karunanidi-Bjss** [reply]
I have configured the cli config but the lambda is still rebuilding the cache on start up, overwriting (or at least trying to) create a json file in the cache directory. Where do I tell the code to use the cache on start up? 

FYI I am using nestjs and typescript

**Reply by B4nan** [reply]
I cant help you with such things without seeing what you are doing.

**Reply by gusgorman2** [reply]
Is there a flag we are supposed to set to tell Mikro ORM to use the prebuilt cache rather than creating it on startup?

**Reply by B4nan** [reply]
See https://mikro-orm.io/docs/guide/advanced#deployment
